src/celt_lpc.c

```diff
@@ -96,7 +96,7 @@ void celt_fir(
          int ord)
 {
    int i,j;
-   opus_val16 rnum[ord];
+   opus_val16 *rnum = calloc(sizeof(opus_val16), ord);
```

Comment:
Why change VLAs to calloc? VLAs are allocated on the stack and calloc allocates on the heap. Usually people prefer using the stack instead of the heap. The other difference is that calloc 0 initializes the memory but this doesn't seem required in these cases.

Reply:
agreed, however after browsing the issues list it seemed some folks had trouble compiling under MSVC due to VLA's. calloc was opted for safety.

Reply:
It's good to support different compilers but I'd prefer some kind of `ifdef` to fallback to heap allocation only on platforms that don't support VLAs like MSCV because supporting them should not negatively affect more modern compilers.
Or maybe the api of these functions could be changed so that the extra buffer is passed in from the outside? I'm worried about allocating too often because it's unclear to me how how often these functions that use VLAs are called (how hot they are).
Changing the api is probably a bigger change than what you want to accomplish in this PR. Maybe something for a maintaner of the repo to decide.

Reply:
Looking at the code more I am confused. `celt_fir` and `celt_iir` are unused functions. They are never called and the project compiles when I remove them.

Reply:
i can revert the VLA related changes if it requires more consideration. i was merely trying to bundle common fixes.

Reply:
It's totally up to you or a maintainer. I'm not an official maintainer, just voicing my opinion on the PR. Also thanks for bundling the fixes.

Reply:
You may want to look into [libspeexdsp](https://github.com/xiph/speexdsp/) and how it handles VLA support.

Reply:
i reverted the VLA related changes. thanks for the feedback wrt libspeex, i might try folding this code back into libspeexdsp to replace it's VAD as an experiment, in which case i would definitely adopt existing conventions.